zipline/data/fx/hdf5.py

```diff
@@ -203,56 +201,55 @@ def get_rates(self, rate, quote, bases, dts):
                 .format(rate, quote)
             )
 
-        # OPTIMIZATION: Row indices correspond to dates, which must be in
+        # OPTIMIZATION: Column indices correspond to dates, which must be in
         # sorted order. Rather than reading the entire dataset from h5, we can
-        # read just the interval from min_row to max_row inclusive
+        # read just the interval from min_col to max_col inclusive
         #
         # However, we also need to handle two important edge cases:
         #
-        #   1. row_ixs contains -1 for dts before the start of self.dts.
-        #   2. col_ixs contains -1 for any currencies we don't know about.
+        #   1. row_ixs contains -1 for any currencies we don't know about.
+        #   2. col_ixs contains -1 for dts before the start of self.dts.
         #
         # If either of the above cases obtains, we want to return NaN for the
         # corresponding output locations.
 
         # We handle (1) by reading raw data into a buffer with one extra
-        # row. When we then apply the row index to permute the raw data into
-        # the correct order, any rows with values of -1 will pull from the
-        # extra row, which will always contain NaN>
+        # column. When we then apply the column index to permute the raw data
```

Comment:
I think this comment got a bit scrambled? `(1)` still refers to the possibility of nonexistent rows, but then we talk about columns here.

Reply:
Now that we use the extra row/column trick for both cases, I reworded the comment.